analyzer/src/funTest/kotlin/curation/ClearlyDefinedPackageCurationProviderFunTest.kt

```diff
@@ -28,35 +28,36 @@ import io.kotest.matchers.shouldNot
 
 import org.ossreviewtoolkit.clients.clearlydefined.ClearlyDefinedService.Server
 import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
 import org.ossreviewtoolkit.utils.spdx.toSpdx
 
 class ClearlyDefinedPackageCurationProviderFunTest : WordSpec({
     "The production server" should {
         val provider = ClearlyDefinedPackageCurationProvider()
 
         "return an existing curation for the javax.servlet-api Maven package" {
-            val identifier = Identifier("Maven:javax.servlet:javax.servlet-api:3.1.0")
+            val packages = createPackagesFromIds("Maven:javax.servlet:javax.servlet-api:3.1.0")
```

Comment:
commit-message: I believe I don't fully get to which plan [1] refers to.
E.g. what are URL based providers and what would be example use cases?
Mind helping me understanding? 

[1] So, as a preparation for ORT to replace the
hard-coded providers [3] with real ones determined based on URLs

Reply:
> Mind helping me understanding?

I'll try: Currently, [this code](https://github.com/oss-review-toolkit/ort/blob/33531c7/model/src/main/kotlin/utils/Extensions.kt#L38-L57) hard-codes e.g. the provider for all `ComponentType.MAVEN` packages to `Provider.MAVEN_CENTRAL`. But that's not correct as not all Maven packages are hosted on Maven Central. To fix that, the code needs to be able to look at the URL of the `RemoteArtifact`, and only if that *really* refers to Maven Central, use `MAVEN_CENTRAL` as the provider.